src/AppInstallerRepositoryCore/AggregatedSource.cpp

```diff
@@ -0,0 +1,72 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+#include "pch.h"
+#include "AggregatedSource.h"
+
+namespace AppInstaller::Repository
+{
+    AggregatedSource::AggregatedSource()
+    {
+        m_details.Name = "AggregatedSource";
+        m_details.IsAggregated = true;
+    }
+
+    const SourceDetails& AppInstaller::Repository::AggregatedSource::GetDetails() const
+    {
+        return m_details;
+    }
+
+    void AggregatedSource::AddSource(std::shared_ptr<ISource> source)
+    {
+        m_sources.emplace_back(std::move(source));
+    }
+
+    SearchResult AggregatedSource::Search(const SearchRequest& request)
+    {
+        SearchResult result;
+
+        for (auto& source : m_sources)
+        {
+            auto oneSourceResult = source->Search(request);
+
+            for (auto& r : oneSourceResult.Matches)
+            {
+                r.SourceName = source->GetDetails().Name;
+                result.Matches.emplace_back(std::move(r));
+            }
+        }
+
+        SortResultMatches(result.Matches);
+
+        if (request.MaximumResults > 0 && result.Matches.size() > request.MaximumResults)
+        {
+            result.Truncated = true;
+            result.Matches.erase(result.Matches.begin() + request.MaximumResults, result.Matches.end());
+        }
+
+        return result;
+    }
+
+    void AggregatedSource::SortResultMatches(std::vector<ResultMatch>& matches)
+    {
+        // TODO: for now just simply prefer winget source.

```

Comment:
If the highest sort priority is source, then no sorting is required.  The sources should be left in their defined order, and so the results will already be good.

The current source already sorts the returned results, first by MatchType, then by Field.  The only other sort option that I have thought of beyond source first is to merge the results in this same sort priority: { MatchType, Field, Source }.  This should allow the results to better represent the search request.